### ISSUE TITLE
"Disable for this line" code action puts the comment on the wrong line (typescript/no-unnecessary-type-parameters)

### ISSUE THREAD

**Issue by zachkirsch** [oxc-project/tsgolint#769]
the `typescript/no-unnecessary-type-parameters` rule targets a different node than it highlights in the IDE. The "disable for this line" code action in VSCode puts the `// oxlint-disable-next-line` comment in the wrong place:

https://github.com/user-attachments/assets/747db30a-11eb-4eb6-ae83-39d2ba8b18b9

**Comment by Sysix** [comment]
I looked into the source code of tsgolint & oxlint & LSP. I could not find a reason why it should "not work". 
Can you confirm that running this in CLI will result into the same issue?

**Comment by zachkirsch** [comment]
> Can you confirm that running this in CLI will result into the same issue?

Sorry, not really following. The issue is that VSCode code action puts the disable directive comment on the wrong line.

**Comment by Sysix** [comment]
I am sorry, I should have explained a bit more :') 
Does the same line/code are shown in CLI for this specific error?
Does putting the disable-line on the "wrong line" silent the error in CLI?
I would love to know if there is a difference between CLI and Editor when it comes to tsgolint diagnostics ignoring lines. :)

**Comment by zachkirsch** [comment]
Ah got it! yeah the CLI is in sync with the video. ie when there's red lines in the IDE I get a failure from the CLI, and when the IDE is clean there's no errors from the CLI.

**Comment by Sysix** [comment]
@camc314 looks like a tsgolint <-> oxlint bug? My first guess would be the `TsgoLintDiagnostic` to `OxcDiagnostic` conversation.

**Comment by zachkirsch** [comment]
thank you @camc314 !

**Comment by wagenet** [comment]
This still reproduces for `typescript/no-unnecessary-type-assertion` on a **multi-line** node, with #20436 present in the build — so the "use the primary span for the disable directive" change doesn't appear to cover this shape. Happy to open a fresh issue if you'd rather track it separately.

The CLI symptom is the same mismatch @Sysix suspected in the diagnostic conversion, just from the other direction: the diagnostic is *reported* at the node start, but suppression is keyed to the line the `as` sits on. When the two are the same line it works; when the node spans lines it does not.

```ts
declare function f(o: object): { id: string };

// single line -- suppressed correctly
// eslint-disable-next-line typescript/no-unnecessary-type-assertion
export const a = f({}) as { id: string };

// multi-line -- NOT suppressed, and the directive is reported unused
// eslint-disable-next-line typescript/no-unnecessary-type-assertion
export const b = f({
	x: 1,
}) as { id: string };
```

```console
$ oxlint --type-aware --report-unused-disable-directives -c oxlint.config.json -- y.ts
y.ts:9:18: error typescript(no-unnecessary-type-assertion): This assertion is unnecessary since it does not change the type of the expression.
y.ts:8:1: warning: Unused eslint-disable directive (no problems were reported).
```

Config is just `{ "plugins": ["typescript"], "rules": { "typescript/no-unnecessary-type-assertion": "error" } }` with a strict tsconfig.

Two things make it worse than a stray diagnostic:

1. The directive is simultaneously reported **unused**, so `--report-unused-disable-directives` tells you to delete a comment that is genuinely needed. The diagnostic is at line 9 and the directive covers line 9, yet both the suppression and the usage accounting disagree with the reported position — in the same run, deterministically (6/6 runs).
2. With `--fix`, the assertion is then removed, so code the author explicitly protected gets rewritten. In our codebase that turned a type-checking file into one that fails `tsc`, because the assertion was load-bearing.

Moving the comment down so it precedes the `}) as ...` line suppresses it, which is what pinned the cause:

```ts
export const b = f({
	x: 1,
// eslint-disable-next-line typescript/no-unnecessary-type-assertion
}) as { id: string };
```

That placement is what a user would never write, and isn't what ESLint semantics imply.

Environment: oxlint 1.81.0, TypeScript 7.1.0-dev.20260901.1. The binary is a local build from a branch carrying one unrelated patch (it flips the `--run-external-code` default); 0c57312 is an ancestor of that build, so #20436 is included.

---

Written by Claude (Claude Code) on @wagenet's behalf and disclosed per Oxc's AI usage policy; the prose is Claude's, not his. Commands and output above were executed and copied verbatim.
